### DISCUSSION TITLE
Strange behavior in UI when deleting

### DISCUSSION THREAD

**Discussion by gigamesh** [supabase/supabase#22910] [Questions] [answered]
I just was prevented from deleting all rows from a table, and received the error "cannot truncate a table referenced in a foreign key constraint"

However, when I selected the currently visible rows (100 at a time), I was able to delete all of them.

This is the relevant part of my schema. I was deleting users and the only foreign key constraint is coming from the Allocation table (which had no rows). Seems to possibly be a bug?

```
model User {
  id          String   @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
...
  allocations Allocation[]
}

model Allocation {
  id        String   @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
  userId    String   @db.Uuid
  user      User     @relation(fields: [userId], references: [id], onDelete: Cascade)
  airdropId String?  @db.Uuid
  airdrop   Airdrop? @relation(fields: [airdropId], references: [id])
 ...
  @@unique([userId, airdropId])
}
```

**Comment by GaryAustin1** [comment, marked as answer]
Postgres truncate is limited in what it can do as it is designed to delete the table very quickly so has limitations.
See here in Notes if interested: https://www.postgresql.org/docs/current/sql-truncate.html
But does seem like an issue on passing the error to the user without explanation or going to a slower delete method.

You might want to generate an issue in supabase/supabase github issues for the dashboard if there is not one there yet.
